ksql-metastore/src/main/java/io/confluent/ksql/metastore/KsqlTable.java

```diff
@@ -62,6 +63,11 @@ public StructuredDataSource cloneWithTimeField(String timestampfieldName) {
                          stateStoreName, isWindowed);
   }
 
+  @Override
+  public QueryId getPersistentQueryId() {
+    return new QueryId("CTAS_" + dataSourceName);
+  }
+
```

Comment:
Same as above, why not in `PhysicalPlanBuilder`?

Reply:
If we put it in these classes, then when we have different persistent queries, i.e., inserts etc, those objects know how to generate their own queryId. Rather than having some other `if(..){..}else if(...)` block elsewhere in the code. i.e., these objects know what the query id should be.